lib/segment/src/index/field_index/full_text_index/posting_list.rs

```diff
@@ -37,23 +40,305 @@ impl PostingList {
         self.list.binary_search(val).is_ok()
     }
 
-    pub fn iter<'a>(
-        &'a self,
-        filter: impl Fn(PointOffsetType) -> bool + 'a,
-    ) -> impl Iterator<Item = PointOffsetType> + 'a {
-        self.list.iter().copied().filter(move |&idx| filter(idx))
+    pub fn iter(&self) -> impl Iterator<Item = PointOffsetType> + '_ {
+        self.list.iter().copied()
     }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct CompressedPostingList {
+    len: usize,
+    last_doc_id: PointOffsetType,
+    data: Vec<u8>,
+    chunks: Vec<CompressedPostingChunk>,
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct CompressedPostingChunk {
+    initial: PointOffsetType,
+    offset: u32,
+}
+
+impl CompressedPostingList {
+    pub fn new(mut posting_list: PostingList) -> Self {
+        if posting_list.list.is_empty() {
+            return Self::default();
+        }
+        let len = posting_list.len();
+        let last_doc_id = *posting_list.list.last().unwrap();
```

Comment:
Is it expected to be obtined before sorting?

Reply:
Removed, thanks!